posthog/api/insight.py

```diff
@@ -284,13 +300,24 @@ def get_last_refresh(self, insight: Insight):
         if should_refresh(self.context["request"]):
             return now()
 
+        dashboard_tile = self.dashboard_tile_from_context(insight, self.context.get("dashboard", None))
+
         result = self.get_result(insight)
         if result is not None:
-            return insight.last_refresh
-        if insight.last_refresh is not None:
-            # Update last_refresh without updating "updated_at" (insight edit date)
-            insight.last_refresh = None
-            insight.save(update_fields=["last_refresh"])
+            if dashboard_tile is not None:
```

Comment:
Here and elsewhere, there shouldn't be need to write `is not None`

```suggestion
            if dashboard_tile:
```